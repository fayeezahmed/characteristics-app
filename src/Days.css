.days {
    display: flex;
}

.day {
    display: flex;
    flex: 1;
    background-color: lightpink;
    list-style: none;
    margin: .1em;
    padding: .3em;
    border-radius: 3px;
    margin: 1px;
}
