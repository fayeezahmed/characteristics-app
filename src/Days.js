import React, { Component } from "react";
import './Days.css';
 
class Day extends Component {
  createDays(day) {
      return <li key={day.toString()} className="day">{day}</li>   
  }
 
  render() {
    var days = this.props.propDays
    var listDays = days.map(this.createDays);
    return (
        // maybe use a table instead? Here as well as traits?
        <ul key="dayofweek" className="days">
            {listDays}
        </ul>
    )
  }
};
 
export default Day;