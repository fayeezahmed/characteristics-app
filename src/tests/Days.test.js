import React from 'react';
import Days from '../Days';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const propDays = ['', 'M', 'T', 'W', 'Th', 'F', 'S', 'Sn'];
  const tree = renderer.create(<Days propDays={propDays}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
