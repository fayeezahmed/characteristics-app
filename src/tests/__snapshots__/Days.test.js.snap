// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="days"
>
  <li
    className="day"
  >
    
  </li>
  <li
    className="day"
  >
    M
  </li>
  <li
    className="day"
  >
    T
  </li>
  <li
    className="day"
  >
    W
  </li>
  <li
    className="day"
  >
    Th
  </li>
  <li
    className="day"
  >
    F
  </li>
  <li
    className="day"
  >
    S
  </li>
  <li
    className="day"
  >
    Sn
  </li>
</ul>
`;
